java/ql/src/Metrics/Summaries/TopJdkApis.qll

```diff
@@ -287,16 +287,19 @@ predicate hasApiName(Callable c, string apiName) {
 }
 
 /** A top JDK API. */
-class TopJdkApi extends SummarizedCallableBase {
+class TopJdkApi extends Callable {
   TopJdkApi() {
+    this.isSourceDeclaration() and
     exists(string apiName |
-      hasApiName(this.asCallable(), apiName) and
+      hasApiName(this, apiName) and
       topJdkApiName(apiName)
     )
   }
 
   /** Holds if this API has a manual summary model. */
-  private predicate hasManualSummary() { this.(SummarizedCallable).hasManualModel() }
+  private predicate hasManualSummary() {
+    exists(SummarizedCallable sc | sc.asCallable() = this and sc.hasManualModel())
```

Comment:
Would it be better to remove the import at line 4 in this file (`private import semmle.code.java.dataflow.FlowSummary`) and to change this line to the below instead? (This would be similar to the updates in the `DataFlowDispatch.qll` file.)
```suggestion
    exists(FlowSummaryImpl::Public::SummarizedCallable sc | sc.asCallable() = this and sc.hasManualModel())
```